[PyTorch] Reporting OOM events to the Pytorch Profiler.

c-odrin opened the pull request:
Summary: Similar to reporting alloc and dealloc events in the PyTorch profiler, we are now reporting Out of Memory events as well. This is useful for performance troubleshooting

Test Plan: Added test_oom_tracing to test/test_profiler.py

Differential Revision: D36268132

robieta reviewed (approved):
LGTM provided the benchmark shows no regression and OSS tests pass.

Thanks a lot for adding this.

c-odrin commented:
[soulitzer](https://github.com/soulitzer) can I get a review on this? Many thanks

c-odrin committed:
[PyTorch] Reporting OOM events to the Pytorch Profiler. (#80050)

Summary:
Pull Request resolved: https://github.com/pytorch/pytorch/pull/80050

Similar to reporting alloc and dealloc events in the PyTorch profiler, we are now reporting Out of Memory events as well. This is useful for performance troubleshooting

Test Plan: Added test_oom_tracing to test/test_profiler.py

Reviewed By: robieta

Differential Revision: D36268132

fbshipit-source-id: 443adac06feaa498796302b2b49073d3e7c8ef91
